active/0003-extensible-resource-deployment-observability-improvements.md

```diff
@@ -0,0 +1,249 @@
+- Performing resource-level or provider-level queries that have no side effects through provider functions in Bicep files or ARM templates, e.g., list secrets of a resource (currently, this is only supported by the az provider).
+
+### Extensibility Provider
+
+An extensibility provider refers to any deployment provider that operates outside the scope of ARM, such as the MS Graph provider and the Kubernetes provider. An extensibility provider deployes extensible resources instead of Azure resources.
+
+### Extensible Resource
+
+An Azure data-plane resource or non-Azure resource declared in a Bicep file or an ARM template. Extensible resources are deployed through extensibility providers.
+
+## Motivation
+
+Users currently encounter a major challenge when deploying extensible resources, as there is inadequate information available when querying the deployments or deployment operation APIs. Key details, such as resource IDs, are not applicable to extensible resources, making it difficult for users to monitor the progress of their deployments. For instance, Azure Portal users attempting to view extensible resource deployment on the deployment details blade will only see an infinite loading animation without any details being displayed. Similarly, Azure PowerShell users tracking deployment progress using the deployment cmdlet and the `-Verbose` switch will receive logs like `VERBOSE: 00:00:00 - Resource '' provisioning status is succeeded`.
+
+To address this issue, the deployments and deployment operations APIs must be updated to include additional information, such as:
+
+- Resource symbolic name
+- Resource type
+- Deployment provider alias
+- Deployment provider name
+- Deployment provider version
+
+This additional data would enable users to identify a deployed extensible resource and map it to the corresponding resource definition in a Bicep file or an ARM template, providing users with comprehensive insights.
+
+## Detailed design
+
+### `Microsoft.Resources/deployments` API changes for PUT and GET
+
+#### Adding `deploymentProviders`
+
+The `Microsoft.Resources/deployments` API will be updated to include a `deploymentProviders` array in both the PUT and GET response bodies. This modification is intended to incorporate information about deployment providers that are involved in executing the parent deployment and nested deployments to offer a more comprehensive deployment view.
+
+> It's worth noting that we cannot use the name `providers` as the property `providers` is already assigned to return Azure Resource Provider information. While an alternative approach could involve renaming the current `providers` property to `resourceProviders` for enhanced accuracy, the solution should be avoided as it constitutes a breaking change.
+
+```diff
+{
+  ...
+  "type": "Microsoft.Resources/deployments",
+  "properties": {
+    ...
+    "providers": [
```

Comment:
What's the consideration to create a new array vs append to this array?

Reply:
Appending to this array means we will be mixing "Azure resource providers" with "extensibility providers", which I think we should avoid due to their conceptually distinct natures.